draft-ietf-mls-protocol.md

```diff
@@ -1504,51 +1504,42 @@ member of the group.
 # Ratchet Tree Operations
 
 The ratchet tree for an epoch describes the membership of a group in that epoch,
-providing public-key encryption keys that can be used to encrypt to subsets of
+providing public-key encryption (HPKE) keys that can be used to encrypt to subsets of
 the group as well as information to authenticate the members.  In order to
 reflect changes to the membership of the group from one epoch to the next,
 corresponding changes are made to the ratchet tree.  In this section, we
 describe the content of the tree and the required operations.
 
 ## Ratchet Tree Evolution
 
-A member of an MLS group advances the key schedule to provide forward secrecy
-and post-compromise security by providing the group with fresh key material to
-be added into the group's shared secret.
-To do so, one member of the group generates fresh key
-material, applies it to their local tree state, and then sends this key material
-to other members in the group via an UpdatePath message (see {{update-paths}}) .
-All other group members then apply the key material in the UpdatePath to their
-own local tree state to derive the group's now-updated shared secret.
+In order to provide forward secrecy and post-compromise security, whenever a
+member initiates an epoch change (i.e., commits; see {{commit}}), they refresh
+all HPKE key pairs in the ratchet tree for which they know the secret keys.
+More precisely, they refresh the key pairs of their leaf and of all nodes on
+their leaf's direct path.
 
-To begin, the generator of the UpdatePath updates its leaf
-KeyPackage and its direct path to the root with new secret values.  The
-HPKE leaf public key within the KeyPackage MUST be derived from a freshly
-generated HPKE secret key to provide post-compromise security.
+The member initiating the epoch change generates the fresh key pairs using the
+following procedure. The procedure is designed in a way that allows to
+efficiently communicate the fresh secret keys to other group members, as
+described in {{update-paths}}. 
 
-The generator of the UpdatePath starts by sampling a fresh random value called
-"leaf_secret", and uses the leaf_secret to generate their leaf HPKE key pair
-(see {{key-packages}}) and to seed a sequence of "path secrets", one for each
-ancestor of its leaf. In this setting,
-path_secret\[0\] refers to the leaf's parent,
-path_secret\[1\] to the parent's parent, and so on. At each step, the path
-secret is used to derive a new secret value for the corresponding
-node, from which the node's key pair is derived.
+* Generate a fresh HPKE key pair for the leaf.
+* Generate a sequence of path secrets, one for each node on the leaf's direct
+path as follows. Sample a fresh random value `path_secret[0]` and compute
```

Comment:
I would split the middle bullet here:

* Generate the leaf key pair
* Generate path secrets
    * `path_secret[0]` is sampled at random
    * `path_secret[n] = DeriveSecret(path_secret[n-1], "path")`
* Derive node key pairs from path secrets

Reply:
Note also that the current version doesn't render quite properly.  "In this context..." isn't rendered as part of the bullet.  You might be able to fix this by indenting that part by 2 spaces.  Or you might have better success putting the code block after the prose.